ihmc-java-toolkit/src/main/java/us/ihmc/tools/thread/MissingThreadTools.java

```diff
@@ -1,82 +1,16 @@
 package us.ihmc.tools.thread;
 
 import us.ihmc.commons.Conversions;
-import us.ihmc.commons.RunnableThatThrows;
-import us.ihmc.commons.exception.DefaultExceptionHandler;
-import us.ihmc.commons.exception.ExceptionHandler;
-import us.ihmc.commons.exception.ExceptionTools;
 import us.ihmc.commons.thread.ThreadTools;
 
-import java.util.concurrent.*;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.LockSupport;
 
 public class MissingThreadTools
 {
-   /**
-    * Guarantees a sleep of a minimum duration in floating point seconds
-    * using {@link LockSupport#parkNanos}. It will always sleep a little too long.
-    * The amount overslept probably varies by system, but it has been observed to
-    * be less than half a millisecond.
-    *
-    * {@link ThreadTools#sleepSeconds} can actually return early because it
-    * cuts off the subnanosecond part, allowing it to undersleep by a nanosecond
-    * at most.
-    *
-    * @param duration to sleep in seconds
-    * @return Exactly how long it actually slept in seconds
-    */
-   public static double sleepAtLeast(double duration)
-   {
-      double startTime = Conversions.nanosecondsToSeconds(System.nanoTime());
-      double amountSlept = 0.0;
-      do
-      {
-         double nextDuration = duration - amountSlept;
-
-         sleep(nextDuration);
-
-         amountSlept = Conversions.nanosecondsToSeconds(System.nanoTime()) - startTime;
-      }
-      while (amountSlept < duration);
-      return amountSlept;
-   }
-
-   /**
-    * Uses {@link LockSupport#parkNanos} to sleep for floating point seconds.
-    * {@link LockSupport#parkNanos} is more accurate than {@link Thread#sleep}.
-    * The requested sleep is guaranteed to be at least as long as the requested
-    * amount and can be up to a nanosecond longer.
-    */
-   public static void sleep(double seconds)
-   {
-      double floatingNanos = seconds * 1e9;
-      long nanoseconds = (long) floatingNanos;
-
-      if (floatingNanos > nanoseconds) // Take nanosecond ceiling instead of floor
-         ++nanoseconds;
-
-      LockSupport.parkNanos(nanoseconds); // More accurate than Thread.sleep
-   }
-
-   public static void sleepMillis(int millis)
-   {
-      try
-      {
-         Thread.sleep(millis);
-      }
-      catch (InterruptedException interruptedException)
-      {
-         // Ignore
-      }
-   }
-
-   public static ThreadFactory createNamedThreadFactory(String prefix, boolean daemon)
-   {
-      boolean includePoolInName = true;
-      boolean includeThreadNumberInName = true;
-      return ThreadTools.createNamedThreadFactory(prefix, includePoolInName, includeThreadNumberInName, daemon, Thread.NORM_PRIORITY);
-   }
-
    public static ResettableExceptionHandlingExecutorService newSingleThreadExecutor(String prefix)
```

Comment:
I didn't move the `ResettableExceptionHandlingExecutorService` stuff because the class says that it's WIP